core/trino-spi/src/main/java/io/trino/spi/security/PasswordAuthenticator.java

```diff
@@ -23,5 +24,16 @@ public interface PasswordAuthenticator
      * @return the authenticated entity
      * @throws AccessDeniedException if not allowed
      */
+    @Deprecated
     Principal createAuthenticatedPrincipal(String user, String password);
+
+    /**
+     * If implemented and authenticated; provide an {@link Identity}.
+     *
+     * @return the authenticated {@link Identity}
+     */
+    default Optional<Identity> createAuthenticatedIdentity(String user, String password)
```

Comment:
What does it mean `Optional.empty()` here?


Same question to other authenticators.

Reply:
I don't like this contract. That means that you introduced a new method to have a smoth transition persiod from one method to the other. However when you remove legacy method, you need to change the method too to remove the `Optional`.

I see two options. Either return exception here that method is not implemented `UnsupportedOperationException` or remove the old method.

I prefer the latter as I believe it will be painful anyway for consumers of that API, defering the pain is not making it easier, but it make it only longer.